base/MAPL_VarSpecMod.F90

```diff
@@ -164,6 +173,44 @@ module MAPL_VarSpecMod
   type(MAPL_VarConnType), pointer :: ConnPtr => null()
 end type MAPL_VarConn
 
+type MAPL_VarServiceProviderType
+  private
+  character(len=ESMF_MAXSTR)               :: SERVICE_NAME
+  character(len=ESMF_MAXSTR)               :: BUNDLE_NAME
+  type(ESMF_FieldBundle)                   :: BUNDLE
+  !ALT currect assumption is the bundle for the provider will be in the import state
+end type MAPL_VarServiceProviderType
+
+type MAPL_VarServiceSubscriberType
+  private
+  character(len=ESMF_MAXSTR)               :: SERVICE_NAME
+  type(ESMF_FieldBundle)                   :: BUNDLE
+  !ALT currect assumption is the bundle for the provider will be in the export state
+  character(len=ESMF_MAXSTR), allocatable      :: VAR_LIST(:)
+end type MAPL_VarServiceSubscriberType
+
+type, public :: MAPL_VarServiceConnectionType
+  private
+  character(len=ESMF_MAXSTR)               :: SERVICE_NAME
+  character(len=ESMF_MAXSTR)               :: PROVIDER_NAME
+  character(len=ESMF_MAXSTR)               :: SUBSCRIBER_NAME
+end type MAPL_VarServiceConnectionType
+
+type, public :: MAPL_VarServiceProviderPtr
+  private
+  type(MAPL_VarServiceProviderType), pointer :: Ptr => null()
+end type MAPL_VarServiceProviderPtr
+
+type, public :: MAPL_VarServiceSubscriberPtr
+  private
+  type(MAPL_VarServiceSubscriberType), pointer :: Ptr => null()
+end type MAPL_VarServiceSubscriberPtr
+
+type, public :: MAPL_VarServiceConnectionPtr
+  private
+  type(MAPL_VarServiceConnectionType), pointer :: Ptr => null()
+end type MAPL_VarServiceConnectionPtr
```

Comment:
Since you need the wrapper types anyway, Vector is probably a better paradigm.  (And the lowest risk FTL component.)    It will require that the basic types MAPL_VarServiceSubscriberType et al will need to be in their own module first.   But that's not so bad of a thing.

I'll also probably push that these adopt a more OO naming style.  (No MAPL prefix, and no "TYPE" suffix.